### ISSUE TITLE
LuaSkin: hs.task:launch() Unable to launch hs.task process: launch path not accessible

### ISSUE THREAD

**Issue by OliverPearmain** [Hammerspoon/hammerspoon#3016]
_If there's a better place to get support please do direct me :-)_

I am trying to execute some shell script from Hammerspoon but I've been unable to get `hs.task` to work. When I attempt to `start` the task I get the error 

> Unable to launch hs.task process: launch path not accessible

I'd be very grafteful for any help. I've tried all sorts unsuccessfully. I am unable to even get the most simple of commands to execute. See example below. 

### Environment
- MacOS Monterey v12.0
- Hammerspoon v0.9.91 (6078)
- Accesibility permission has been granted/enabled under System Preferences

### init.lua

```
hs.hotkey.bind({"cmd", "alt", "ctrl"}, "W", function()
    hs.task.new("say hello", nil):start()
end)
```

### Console Output
```
2021-12-12 20:59:11: Welcome to the Hammerspoon Console!
You can run any Lua code in here.

2021-12-12 20:59:11: -- Lazy extension loading enabled
2021-12-12 20:59:11: -- Loading ~/.hammerspoon/init.lua
2021-12-12 20:59:11: -- Loading extension: hotkey
2021-12-12 20:59:11: 20:59:11     hotkey: Enabled hotkey ⌘⌃⌥W
2021-12-12 20:59:11: -- Done.
2021-12-12 20:59:16: -- Loading extension: task
2021-12-12 20:59:16: 20:59:16 ** Warning:   LuaSkin: hs.task:launch() Unable to launch hs.task process: launch path not accessible
```

**Comment by cmsj** [comment]
`hs.task` is very unforgiving, it's intended to be used with high precision about what you want. It's perfectly fine to use it for this, but you'll need to do something like this instead:

```lua
hs.task.new("/usr/bin/say", nil, {"hello"}):start()
```

So firstly that means passing the full path to the `say` binary, and then passing the individual arguments for `say` as a table (with a `nil` in there too because in this case we don't want to pass a callback function for when the task has completed).

FWIW the full docs for `hs.task.new()` are [here](http://www.hammerspoon.org/docs/hs.task.html#new)

**Comment by cmsj** [comment]
Just as a further thought - the easy way to do this would be `hs.execute("say hello")` which (unlike `hs.task`) hands the entire string you enter, off to your shell to handle as it would normally. The downside of `hs.execute()` is that it will block Hammerspoon until the command has finished executing. `hs.task` is great because it runs the process asynchronously without blocking Hammerspoon (hence the `completionFn` argument you can pass to it, to be able to do some useful Lua work after the command has finished).

I hope that helps!

**Comment by OliverPearmain** [comment]
Aha, you legend, thank you so much for taking the time to reply. 🙇 🙇 🙇 

All is now working - provided I use the full path, and omit spaces (passing args instead) just like you've suggested.

An example, in case it helps anyone else...
```
	hs.task.new("/bin/sh", function(...)
		print("exit", hs.inspect(table.pack(...)))
	end, function(ud, ...)
		print("stream", hs.inspect(table.pack(...)))
		return true
	end, { "/path/to/my/script" }):start()
``` 

I had discovered and tried `hs.execute` also but the scripts in question are fairly long running and I found that Hammerspoon would hang for some time, just like you insinuated. 

Thanks so much again. I appreciate you :-)
